pkg/specgen/generate/namespaces.go

```diff
@@ -297,7 +297,14 @@ func namespaceOptions(s *specgen.SpecGenerator, rt *libpod.Runtime, pod *libpod.
 
 	postConfigureNetNS := needPostConfigureNetNS(s)
 
+	// Network
 	switch s.NetNS.NSMode {
+	case specgen.Host:
+		_, expose, err := createPortMappings(s, imageData)
+		if err != nil {
+			return nil, err
+		}
+		toReturn = append(toReturn, libpod.WithExposedPorts(expose))
```

Comment:
I don't like this, what makes host special compared to container/pod? I would assume docker treats all the same?

Logically it seems it would be much cleaner to pass the ports outside of this switch for all net modes and then remove the expose ports from WithNetNS() to simply there.

Reply:
Ok it seems to be inconsistent on the docker side as well
```
docker run --network container:c1 --name c2 --expose 80/tcp  -d alpine sleep 1000
docker: Error response from daemon: conflicting options: port exposing and the container type network mode.
See 'docker run --help'.
[root@fedora ~]# docker run --network host --name c2 --expose 80/tcp  -d alpine sleep 1000
cfec8194400a44235ef707f7ee838a89c973fed6a20e94528ee12d3793a99f62
````

However at the same time using an image that contains exposed ports is allowed
```
# docker run --network container:c1 --name c2   -d nginx sleep 1000

# docker inspect --format "{{json .Config.ExposedPorts}}"  c2
{"80/tcp":{}}
```

So overall I think we should allow the exposed ports for all network modes.

Reply:
Fixed